### PR TITLE
Improve typing coverage

### LINKED ISSUE #87: [Enhancement] Improve Typing Support / Ensure every function has return type
One of the goals for v1 release was to improve typing support. This issue is only concerned return types.
I wanted to improve the type support so I wrote a script to parse python code and filter out function definitions and class method definitions to find out what needs typing. In the process, I figured that this would be quite useful to add as a test.

## Suggestion

Add a test to ensure that every function has a return type.

This isn't to be confused with type verification, e.g. [bad return types](https://google.github.io/pytype/errors.html#bad-return-type).

## Special Cases

In some cases we may want to avoid adding a return type, a decorator could be added onto the function to make it ignored by the test.

## An exhaustive list of functions that are missing return types:

```
Missing types:
stable_baselines3/common/vec_env/dummy_vec_env.py::DummyVecEnv::step_async
stable_baselines3/common/vec_env/dummy_vec_env.py::DummyVecEnv::step_wait
stable_baselines3/common/vec_env/dummy_vec_env.py::DummyVecEnv::seed
stable_baselines3/common/vec_env/dummy_vec_env.py::DummyVecEnv::reset
stable_baselines3/common/vec_env/dummy_vec_env.py::DummyVecEnv::close
stable_baselines3/common/vec_env/dummy_vec_env.py::DummyVecEnv::render
stable_baselines3/common/vec_env/dummy_vec_env.py::DummyVecEnv::_save_obs
stable_baselines3/common/vec_env/dummy_vec_env.py::DummyVecEnv::_obs_from_buf
stable_baselines3/common/vec_env/dummy_vec_env.py::DummyVecEnv::get_attr
stable_baselines3/common/vec_env/dummy_vec_env.py::DummyVecEnv::set_attr
stable_baselines3/common/vec_env/dummy_vec_env.py::DummyVecEnv::env_method
stable_baselines3/common/vec_env/dummy_vec_env.py::DummyVecEnv::_get_target_envs
stable_baselines3/common/logger.py::SeqWriter::write_sequence
stable_baselines3/common/vec_env/vec_normalize.py::VecNormalize::__getstate__
stable_baselines3/common/vec_env/vec_normalize.py::VecNormalize::__setstate__
stable_baselines3/common/vec_env/vec_normalize.py::VecNormalize::set_venv
stable_baselines3/common/vec_env/vec_normalize.py::VecNormalize::step_wait
stable_baselines3/common/vec_env/vec_normalize.py::VecNormalize::_update_reward
stable_baselines3/common/vec_env/vec_normalize.py::VecNormalize::normalize_obs
stable_baselines3/common/vec_env/vec_normalize.py::VecNormalize::normalize_reward
stable_baselines3/common/vec_env/vec_normalize.py::VecNormalize::unnormalize_obs
stable_baselines3/common/vec_env/vec_normalize.py::VecNormalize::unnormalize_reward
stable_baselines3/common/vec_env/vec_normalize.py::VecNormalize::get_original_obs
stable_baselines3/common/vec_env/vec_normalize.py::VecNormalize::get_original_reward
stable_baselines3/common/vec_env/vec_normalize.py::VecNormalize::reset
stable_baselines3/common/utils.py::check_for_correct_spaces
stable_baselines3/td3/policies.py::TD3Policy::forward
stable_baselines3/common/cmd_util.py::make_vec_env
stable_baselines3/common/cmd_util.py::make_atari_env
stable_baselines3/common/buffers.py::BaseBuffer::sample
stable_baselines3/common/buffers.py::BaseBuffer::_get_samples
stable_baselines3/common/vec_env/vec_frame_stack.py::VecFrameStack::step_wait
stable_baselines3/common/vec_env/vec_frame_stack.py::VecFrameStack::reset
stable_baselines3/common/vec_env/vec_frame_stack.py::VecFrameStack::close
stable_baselines3/common/monitor.py::Monitor::close
stable_baselines3/common/atari_wrappers.py::MaxAndSkipEnv::reset
stable_baselines3/common/evaluation.py::evaluate_policy
stable_baselines3/common/policies.py::BasePolicy::forward
stable_baselines3/common/policies.py::BasePolicy::load_from_vector
stable_baselines3/common/vec_env/vec_video_recorder.py::VecVideoRecorder::reset
stable_baselines3/common/vec_env/vec_video_recorder.py::VecVideoRecorder::start_video_recorder
stable_baselines3/common/vec_env/vec_video_recorder.py::VecVideoRecorder::_video_enabled
stable_baselines3/common/vec_env/vec_video_recorder.py::VecVideoRecorder::step_wait
stable_baselines3/common/vec_env/vec_video_recorder.py::VecVideoRecorder::close_video_recorder
stable_baselines3/common/vec_env/vec_video_recorder.py::VecVideoRecorder::close
stable_baselines3/common/callbacks.py::EvalCallback::_init_callback
stable_baselines3/common/off_policy_algorithm.py::OffPolicyAlgorithm::_setup_model
stable_baselines3/common/off_policy_algorithm.py::OffPolicyAlgorithm::save_replay_buffer
stable_baselines3/common/off_policy_algorithm.py::OffPolicyAlgorithm::load_replay_buffer
stable_baselines3/common/vec_env/util.py::copy_obs_dict
stable_baselines3/common/vec_env/util.py::dict_to_obs
stable_baselines3/common/vec_env/util.py::obs_space_info
stable_baselines3/common/vec_env/vec_check_nan.py::VecCheckNan::step_async
stable_baselines3/common/vec_env/vec_check_nan.py::VecCheckNan::step_wait
stable_baselines3/common/vec_env/vec_check_nan.py::VecCheckNan::reset
stable_baselines3/common/vec_env/vec_check_nan.py::VecCheckNan::_check_val
stable_baselines3/common/vec_env/subproc_vec_env.py::_worker
stable_baselines3/common/vec_env/subproc_vec_env.py::SubprocVecEnv::step_async
stable_baselines3/common/vec_env/subproc_vec_env.py::SubprocVecEnv::step_wait
stable_baselines3/common/vec_env/subproc_vec_env.py::SubprocVecEnv::seed
stable_baselines3/common/vec_env/subproc_vec_env.py::SubprocVecEnv::reset
stable_baselines3/common/vec_env/subproc_vec_env.py::SubprocVecEnv::close
stable_baselines3/common/vec_env/subproc_vec_env.py::SubprocVecEnv::get_attr
stable_baselines3/common/vec_env/subproc_vec_env.py::SubprocVecEnv::set_attr
stable_baselines3/common/vec_env/subproc_vec_env.py::SubprocVecEnv::env_method
stable_baselines3/common/vec_env/subproc_vec_env.py::SubprocVecEnv::_get_target_remotes
stable_baselines3/common/vec_env/subproc_vec_env.py::_flatten_obs
stable_baselines3/common/noise.py::VectorizedActionNoise::base_noise
stable_baselines3/common/vec_env/base_vec_env.py::VecEnv::reset
stable_baselines3/common/vec_env/base_vec_env.py::VecEnv::step_async
stable_baselines3/common/vec_env/base_vec_env.py::VecEnv::step_wait
stable_baselines3/common/vec_env/base_vec_env.py::VecEnv::close
stable_baselines3/common/vec_env/base_vec_env.py::VecEnv::get_attr
stable_baselines3/common/vec_env/base_vec_env.py::VecEnv::set_attr
stable_baselines3/common/vec_env/base_vec_env.py::VecEnv::env_method
stable_baselines3/common/vec_env/base_vec_env.py::VecEnv::step
stable_baselines3/common/vec_env/base_vec_env.py::VecEnv::render
stable_baselines3/common/vec_env/base_vec_env.py::VecEnv::unwrapped
stable_baselines3/common/vec_env/base_vec_env.py::VecEnv::getattr_depth_check
stable_baselines3/common/vec_env/base_vec_env.py::VecEnv::_get_indices
stable_baselines3/common/vec_env/base_vec_env.py::VecEnvWrapper::step_async
stable_baselines3/common/vec_env/base_vec_env.py::VecEnvWrapper::reset
stable_baselines3/common/vec_env/base_vec_env.py::VecEnvWrapper::step_wait
stable_baselines3/common/vec_env/base_vec_env.py::VecEnvWrapper::seed
stable_baselines3/common/vec_env/base_vec_env.py::VecEnvWrapper::close
stable_baselines3/common/vec_env/base_vec_env.py::VecEnvWrapper::render
stable_baselines3/common/vec_env/base_vec_env.py::VecEnvWrapper::get_images
stable_baselines3/common/vec_env/base_vec_env.py::VecEnvWrapper::get_attr
stable_baselines3/common/vec_env/base_vec_env.py::VecEnvWrapper::set_attr
stable_baselines3/common/vec_env/base_vec_env.py::VecEnvWrapper::env_method
stable_baselines3/common/vec_env/base_vec_env.py::VecEnvWrapper::__getattr__
stable_baselines3/common/vec_env/base_vec_env.py::VecEnvWrapper::_get_all_attributes
stable_baselines3/common/vec_env/base_vec_env.py::VecEnvWrapper::getattr_recursive
stable_baselines3/common/vec_env/base_vec_env.py::VecEnvWrapper::getattr_depth_check
stable_baselines3/common/vec_env/base_vec_env.py::CloudpickleWrapper::__getstate__
stable_baselines3/common/vec_env/base_vec_env.py::CloudpickleWrapper::__setstate__
stable_baselines3/dqn/dqn.py::DQN::_on_step
stable_baselines3/common/base_class.py::BaseAlgorithm::load
```

### PR DESCRIPTION
**Pull request by araffin**
## Description

Addresses part of #1

Done:
- [x] most VecEnv and wrappers

Missing:
- [x] Vec video recorder

## Motivation and Context

- [ ] I have raised an issue to propose this change ([required](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) for new features and bug fixes)

closes #87

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist:

- [x] I've read the [CONTRIBUTION](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) guide (**required**)
- [x] I have updated the changelog accordingly (**required**).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.
- [x] I have reformatted the code using `make format` (**required**)
- [x] I have checked the codestyle using `make check-codestyle` and `make lint` (**required**)
- [x] I have ensured `make pytest` and `make type` both pass. (**required**)

Note: we are using a maximum length of 127 characters per line

The remaining method without a return type are (2 because they are abstract method and one because pytype does not play nicely with it):
```
stable_baselines3/common/policies.py::BaseModel::forward
stable_baselines3/common/buffers.py::BaseBuffer::sample
stable_baselines3/common/save_util.py::open_path
```

### CODE REVIEW

File: stable_baselines3/common/sb2_compat/rmsprop_tf_like.py [hunk 1]
```diff
@@ -42,7 +44,16 @@ class RMSpropTFLike(Optimizer):
 
     """
 
-    def __init__(self, params, lr=1e-2, alpha=0.99, eps=1e-8, weight_decay=0, momentum=0, centered=False):
+    def __init__(
```
Comment (Miffyli) [inline stable_baselines3/common/sb2_compat/rmsprop_tf_like.py:47]: Could clean the typing info from docstrings at the same time?

**Comment by Miffyli** [comment]
Looked over the changes and they seem to check out, just a comment on if docstrings should be checked at the same time now that the same files being modified.

File: stable_baselines3/common/sb2_compat/rmsprop_tf_like.py (hunk 1 above)
Comment (araffin) [reply stable_baselines3/common/sb2_compat/rmsprop_tf_like.py:47]: yes, I did not know for that one, as it is a clone of PyTorch code. If we want to compare it in future version.
But  will do, so we have only one codestyle.

Comment (Miffyli) [reply stable_baselines3/common/sb2_compat/rmsprop_tf_like.py:47]: Ah right, good point. It is good at is :)

**Review by Miffyli — APPROVED** [review]
If no other docstrings are about to be modified, LGTM!

**Comment by araffin** [comment]
> If no other docstrings are about to be modified, LGTM!

At the end, it was not much so I did the changes. I also improved the error messages to address #180.

**Pull request merged**